src/toast/todmap/mapmaker.py

```diff
@@ -1585,9 +1585,9 @@ def get_detweights(self, data):
         self.detweights = []
         for obs in data.obs:
             tod = obs["tod"]
-            try:
+            if "noise" in obs:
```

Comment:
Do you disapprove of the "easier ask for forgiveness than permission" programming or the fact that the classes of acceptable exceptions were not listed?

Reply:
Obviously try / except is a standard python idiom.  If you would prefer to use that and list the key error exception that is fine.  I guess it is a matter of preference.  We already know exactly what we want to test (whether the key is in the dictionary), so why not just check that?